share/wake/lib/prim/prim.wake

```diff
@@ -0,0 +1,401 @@
+    (\_ \_ \_ prim "powm")
+
+export def prim_print: String => String => Unit =
+    (\_ \_ prim "print")
+
+export def prim_quote: String => String =
+    (\_ prim "quote")
+
+export def prim_rcat: RegExp =
+    (prim "rcat")
+
+export def prim_rcmp: RegExp => RegExp => Order =
+    (\_ \_ prim "rcmp")
+
+export def prim_re2: String => Result RegExp String =
+    (\_ prim "re2")
+
+export def prim_re2str: RegExp => String =
+    (\_ prim "re2str")
+
+export def prim_read: String => Result String String =
+    (\_ prim "read")
+
+export def prim_relative: String => String => String =
+    (\_ \_ prim "relative")
+
+export def prim_replace: RegExp => String => String => String =
+    (\_ \_ \_ prim "replace")
+
+export def prim_root: Integer => Integer => Integer =
+    (\_ \_ prim "root")
+
+export def prim_sNFC: String => String =
+    (\_ prim "sNFC")
+
+export def prim_sNFKC: String => String =
+    (\_ prim "sNFKC")
+
+export def prim_scaseNFKC: String => String =
+    (\_ prim "scaseNFKC")
+
+export def prim_scmp: String => String => Order =
+    (\_ \_ prim "scmp")
+
+export def prim_search_path: String => String => String =
+    (\_ \_ prim "search_path")
+
+export def prim_shell_str: String => String =
+    (\_ prim "shell_str")
+
+export def prim_shl: Integer => Integer => Integer =
+    (\_ \_ prim "shl")
+
+export def prim_shr: Integer => Integer => Integer =
+    (\_ \_ prim "shr")
+
+export def prim_simplify: String => String =
+    (\_ prim "simplify")
+
+export def prim_sources: String => RegExp => List String =
+    (\_ \_ prim "sources")
+
+export def prim_stack: Unit => List String =
+    (\_ prim "stack")
+
+export def prim_str: Integer => Integer => String =
+    (\_ \_ prim "str")
+
+export def prim_str2bin: String => Integer =
+    (\_ prim "str2bin")
+
+export def prim_str2code: String => Integer =
+    (\_ prim "str2code")
+
+export def prim_strlen: String => Integer =
+    (\_ prim "strlen")
+
+export def prim_sub: Integer => Integer => Integer =
+    (\_ \_ prim "sub")
+
+# export def prim_tget: Target => (Target => a) => a =
+#     (\_ \_ prim "tget")
+
+# export def prim_tnew: String => Integer => Target =
+#     (\_ \_ prim "tnew")
+
+export def prim_tokenize: RegExp => String => List String =
+    (\_ \_ prim "tokenize")
+
+export def prim_true: a => Boolean =
+    (\_ prim "true")
+
+export def prim_uname: Pair String String =
+    (prim "uname")
+
+export def prim_unlink: String => Unit =
+    (\_ prim "unlink")
+
+export def prim_unreachable: String => a =
+    (\_ prim "unreachable")
+
+export def prim_use: a => a =
+    (\_ prim "use")
+
+export def prim_vcat: String =
+    (prim "vcat")
+
+export def prim_version: String =
+    (prim "version")
+
+# export def prim_vget: Array a => Integer => a =
+#     (\_ \_ prim "vget")
+
+# export def prim_vnew: Integer => Array a =
+#     (\_ prim "vnew")
+
+# export def prim_vset: Array a => Integer => a => Unit =
+#     (\_ \_ \_ prim "vset")
```

Comment:
The problem with `Array a` is that its a primitive type so we can't define it in wake but it has a parameter so we don't define it like `Integer` either.

It might be sufficient to create an unconstructable array type so that `prim_vnew` is the only way to construct it but I'm not sure how that would go exactly. I'd rather just drop these than try to shove it in.